packages/compass-saved-aggregations-queries/src/components/saved-item-card.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import {
+  Card,
+  css,
+  useDefaultAction,
+  Badge,
+  Icon,
+  Subtitle,
+  spacing,
+  uiColors,
+  useFocusState,
+  useHoverState,
+  mergeProps,
+  FocusState,
+  Menu,
+  MenuItem,
+  IconButton,
+} from '@mongodb-js/compass-components';
+import { formatDate } from '../util/format-date';
+
+type Action = 'open' | 'delete' | 'copy' | 'edit';
+
+type SavedItemCardProps = {
+  id: string;
+  type: 'query' | 'aggregation';
+  name: string;
+  database: string;
+  collection: string;
+  lastModified: number;
+  onAction(actionName: Action): void;
```

Comment:
just a question,  is this the most idiomatic way nowadays? what is the advantage of doing this vs `onOpen?`, `onDelete?`. Should we follow this pattern elsewhere too / is it something you would recommend to do more consistently?

Reply:
Not that it's idiomatic, it's just a way to do it that I find convenient for the following reasons: it's strictly typed (so it's not that we can mistype something and not handle an action), allows to have all logic that touches plugin state / appRegistry passed through one callback prop (see [sidebar navigation](https://github.com/mongodb-js/compass/blob/4424f3399d85313a787ba0e39d8c7bc9cae1c6fd/packages/compass-sidebar/src/components/sidebar-databases-navigation/index.js#L38-L71) for example where the amount of possible actions is bigger), and reduces the amount of props that you would need to pass around inside the component.

I don't have a strong preference here, so if you find separate callback props a better choice of API here, I'll change it.

Generally speaking I don't think we can have a convention that we apply to every case as the cases might be very different

Reply:
Ok makes sense, thanks!